src/casbin_actor.rs

```diff
@@ -0,0 +1,70 @@
+use actix::prelude::*;
+use casbin::prelude::*;
+use casbin::Result;
+use std::sync::Arc;
+
+pub enum CasbinCmd {
+    Enforce(Vec<String>),
+    AddPolicy(Vec<String>),
+    AddPolicies(Vec<Vec<String>>),
+    RemovePolicy(Vec<String>),
+    RemovePolicies(Vec<Vec<String>>),
+    RemoveFilteredPolicy(usize, Vec<String>),
+}
+
+impl Message for CasbinCmd {
+    type Result = Result<bool>;
+}
+
+pub struct CasbinActor {
+    enforcer: Option<Arc<async_std::sync::RwLock<Enforcer>>>,
+}
+
+impl CasbinActor {
+    pub async fn new<M: TryIntoModel, A: TryIntoAdapter>(m: M, a: A) -> Addr<CasbinActor> {
+        let enforcer: Enforcer = Enforcer::new(m, a).await.unwrap();
+        Supervisor::start(|_| CasbinActor {
+            enforcer: Some(Arc::new(async_std::sync::RwLock::new(enforcer))),
+        })
+    }
+}
+
+impl Actor for CasbinActor {
+    type Context = Context<Self>;
+}
+
+impl Supervised for CasbinActor {
+    fn restarting(&mut self, _: &mut Self::Context) {
+        self.enforcer.take();
+    }
+}
+
+impl Handler<CasbinCmd> for CasbinActor {
+    type Result = ResponseActFuture<Self, Result<bool>>;
+
+    fn handle(&mut self, msg: CasbinCmd, _: &mut Self::Context) -> Self::Result {
+        let e = match &self.enforcer {
+            Some(x) => x,
+            None => panic!("Enforcer needed"),
```

Comment:
insead of panic, we can maybe call `ctx.stop()`

Reply:
because we have implemented `Supervisor`, it''ll restart the actor and call `started` again.

so if we put enforcer creation inside `started`, once an error happens, we will be able to create a fresh enforcer.

Like psql disconnected etc...